src/test/java/dev/cdevents/CDEventsTest.java

```diff
@@ -215,6 +336,33 @@ void createTaskRunFinishedEventAsCloudEvent() {
         assertThat(ceDataJson).isEqualTo(cdEventJson);
     }
 
+    @Test
+    void createTaskRunFinishedEventWithOptionalFieldsUnset() throws IOException {
+        InputStream inputStream = getClass().getResourceAsStream("/taskrun_finished_optional.json");
+        TaskRunFinishedCDEvent expectedCDEvent = objectMapper.readValue(inputStream, TaskRunFinishedCDEvent.class);
+
+        TaskRunFinishedCDEvent cdEvent =  new TaskRunFinishedCDEvent();
+        cdEvent.setSource(URI.create("http://dev.cdevents"));
+
+        cdEvent.setSubjectId("/dev/task/run/subject");
+        cdEvent.setSubjectPipelineRunId("/dev/pipeline/run/subject");
+
+        String cdEventJson = CDEvents.cdEventAsJson(cdEvent);
+        CloudEvent ceEvent = CDEvents.cdEventAsCloudEvent(cdEvent);
```

Comment:
Do we really need to go through the cloudevent binding in this test?
It does not harm but I thought that would be covered already by previous tests. 
It's fine to keep it if you deem it relevant here.

Reply:
Yes to validate the CDEvent against schema, I need to invoke `CDEvents.cdEventAsCloudEvent(cdEvent);` 

But I think I can move validation part to the `CDEvents.cdEventAsJson(cdEvent);` method itself as `cdEventAsJson()` is invoked from `cdEventAsCloudEvent()` to set the data in the CloudEvent. So that I can remove cloudevent binding from this test also.